src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -1177,34 +1178,6 @@ HasComplexRangeTableType(Query *queryTree)
 }
 
 
-/*
- * ExtractRangeTableIndexWalker walks over a join tree, and finds all range
- * table indexes in that tree.
- */
-bool
-ExtractRangeTableIndexWalker(Node *node, List **rangeTableIndexList)
-{
-	bool walkerResult = false;
-	if (node == NULL)
-	{
-		return false;
-	}
-
-	if (IsA(node, RangeTblRef))
-	{
-		int rangeTableIndex = ((RangeTblRef *) node)->rtindex;
-		(*rangeTableIndexList) = lappend_int(*rangeTableIndexList, rangeTableIndex);
```

Comment:
We probably shouldn't try to merge this function in because it does something quite different (returns indexes and does not descend into query).

Reply:
it makes sense. Another question is, should I move that function to util file as it is, without merging it ?